src/app/models/evt-models.ts

```diff
@@ -331,3 +331,60 @@ export class Word extends GenericElement {
 export class Deletion extends GenericElement {
     rend: string;
 }
+
+export class MsDesc extends GenericElement {
+    id? : string;
+    msIdentifier: MsIdentifier;
+    msContents?: MsContents;
+    physDesc?: PhysDesc;
+    history?: History;
+    msPart?: MsPart;
+}
+
+export class MsIdentifier extends GenericElement {
+    id?: string;
```

Comment:
Why is the id optional?

Reply:
I would also set it as mandatory and create one (maybe from the XML path) if none exists in data source

Reply:
The getID() function already exists